### ISSUE TITLE
OrderBy (and others) not working with aliased fields

### ISSUE THREAD

**Issue by cdomigan** [Happyr/Doctrine-Specification#127]
The OrderBy Query Modifier (and others) does not support using aliased fields.

`$qb->addOrderBy(sprintf('%s.%s', $dqlAlias, $this->field), $this->order);`

This line enforces a **table.field** regardless of the dqlAlias passed in. Whereas I will often want to order on an aliased field (with no dot).

**Comment by KyHikari** [comment]
Same problem for the comparisons on aliased fields.

**Comment by Nyholm** [comment]
Let's see if we can support aliased fields. Can you give an example query using specs that you would like to work?

**Comment by mitrae** [comment]
@Nyholm the simplest way would to not add alias if it is empty
'' == $dqlAlias ?  $this->field : sprintf('%s.%s', $dqlAlias, $this->field)

in this case i can add some complex field (with SUM, CASE, etc)  into select and use it later in specs

**Comment by peter-gribanov** [comment]
@mitrae this problem will be solved #169 in the next major release.

**Comment by peter-gribanov** [comment]
Now you can order by aliases:
```php
// DQL: SELECT e, e.posts_count + e.likes_count HIDDEN score FROM User e ORDER BY score
$spec = Spec::andX(
    Spec::addSelect(
        Spec::selectHiddenAs(
            Spec::add(Spec::field('posts_count'), Spec::field('likes_count')),
            'score'
        )
    ),
    Spec::orderBy(
        Spec::alias('score')
    )
);
```
